src/core/border_router/routing_manager.cpp

```diff
@@ -623,6 +621,8 @@ void RoutingManager::StartRouterSolicitationDelay(void)
 
     uint32_t randomDelay;
 
+    mInfraIfRouterAdvMessage.SetToDefault();
```

Comment:
@jwhui Current implementation will start Router Discovery only when all discovered on-link prefixes get expired. That means we will not be able to forget the RA header if:
1. The RA daemon crashed before sending RAs with zero Router Lifetime.
2. There are multiple RA daemons that are both advertising.
3. The crashed RA daemon never come back.

We need to rely on the `STALE_RA_TIME` approach to actively detect the liveness of RA daemons to address this issue.